test/unit/assets/x509/build-testcases.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python
+
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# build-testcases.py: generate some bogus X.509 testcases for sigstore's
+# unit tests.
+#
+# These testcases should already be checked-in; you can re-generate them
+# (with entirely new key material) using:
+#
+#  python build-testcases.py
```

Comment:
This doesn't need to be done in this PR, but would it seem reasonable for this to run as part of the `Makefile` (and if it changes files, breaks CI?).

That'd ensure that this file doesn't get out of sync with the checked in certs and it keeps working with future `cryptography` versions.

Reply:
Yeah, good idea -- I should probably make it use fixed keys and times as well, to ensure that it's reproducible.

Reply:
(Let's create a separate issue to track this, since it isn't a blocker.)

Reply:
I think I have this working already so I'll push it up. If you feel it needs more work, feel free to revert it back out.

Reply:
[151d4f0](https://github.com/sigstore/sigstore-python/pull/632/commits/151d4f034038a29056380b015352782ed4535f55)

Reply:
Cool, will take a look shortly 🙂

Reply:
Looks great, thank you for breaking the keys and other nondeterminism out!